advanced/kernel-output/README.md

```diff
@@ -1,149 +1,91 @@
-    SimplifiedOutputArea.execute(code, this._outputarea, this._session)
-      .then((msg: KernelMessage.IExecuteReplyMsg) => {
-        console.log(msg);
-      })
-      .catch(reason => console.error(reason));
-  }
+1.  the JupyterLab plugin that activates all the extension components and connects
+    them to the main `Jupyterlab` application via commands, launcher and menu
+    items,
+2.  a panel that contains the extension logic and UI elements to interact with it.
 
-  protected onCloseRequest(msg: Message): void {
-    super.onCloseRequest(msg);
-    this.dispose();
-  }
+The first part is contained in the `index.ts` file and the second in `panel.ts`.
 
-  get session(): IClientSession {
-    return this._session;
-  }
-
-  private _session: ClientSession;
-  private _outputarea: SimplifiedOutputArea;
-  private _outputareamodel: OutputAreaModel;
-}
-```
+In the following sections, the logic will be first described. It is
+followed by the visual element creation.
 
 ## Initializing a Kernel Session
 
-The first thing that we want to focus on is the `ClientSession` that is
-stored in the private `_session` variable:
+To interact with a kernel, you first need to create a `ClientSession`
+object ([see the documentation](https://jupyterlab.github.io/jupyterlab/apputils/classes/clientsession.html)).
+Here it is stored in the private `_session` variable:
 
 ```ts
-// src/panel.ts#L74-L74
+// src/panel.ts#L71-L71
 
 private _session: ClientSession;
 ```
 
-A ClientSession handles a single kernel session. The session itself (not yet
+A `ClientSession` handles a single kernel session. The session itself (not yet
 the kernel) is started with these lines:
 
 ```ts
-// src/panel.ts#L34-L40
-
-let path = './console';
+// src/panel.ts#L32-L35
 
 this._session = new ClientSession({
   manager: manager.sessions,
-  path,
-  name: 'Tutorial'
+  name: 'Example'
 });
 ```
 
-A kernel is initialized with this line:
+The private session variable is exposed as read-only for other users
+through a getter method:
 
 ```ts
-// src/panel.ts#L49-L49
+// src/panel.ts#L49-L51
 
-void this._session.initialize();
+get session(): IClientSession {
+  return this._session;
+}
 ```
 
-In case that a session has no predefined favourite kernel, a popup will be
-started that asks the user which kernel should be used. Conveniently, this can
-also be an already existing kernel, as we will see later.
+Once you have created a session, the associated kernel can be initialized
+with this line:
+
+```ts
+// src/panel.ts#L44-L46
+
+this._session.initialize().catch(reason => {
+  console.error(`Fail to initialize session in ExamplePanel.\n${reason}`);
+});
+```
 
-The following three methods add functionality to cleanly dispose of the session
-when we close the panel, and to expose the private session variable such that
-other users can access it.
+When a session has no predefined favourite kernel, a dialog will request the user to choose a kernel to start. Conveniently, this can
```

Comment:
We could use "preferred" like in the kernel selection dialog:

```suggestion
When a session has no predefined preferred kernel, a dialog will request the user to choose a kernel to start. Conveniently, this can
```

Reply:
Indeed it will be more coherent.